cdap-app-fabric/src/main/java/co/cask/cdap/app/mapreduce/MapReduceMetricsInfo.java

```diff
@@ -0,0 +1,217 @@
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package co.cask.cdap.app.mapreduce;
+
+import co.cask.cdap.api.dataset.lib.cube.TagValue;
+import co.cask.cdap.api.dataset.lib.cube.TimeValue;
+import co.cask.cdap.api.metrics.MetricDataQuery;
+import co.cask.cdap.api.metrics.MetricStore;
+import co.cask.cdap.api.metrics.MetricTimeSeries;
+import co.cask.cdap.api.metrics.MetricType;
+import co.cask.cdap.app.metrics.MapReduceMetrics;
+import co.cask.cdap.common.conf.Constants;
+import co.cask.cdap.proto.Id;
+import co.cask.cdap.proto.MRJobInfo;
+import co.cask.cdap.proto.MRTaskInfo;
+import co.cask.cdap.proto.ProgramType;
+import com.google.common.base.Preconditions;
+import com.google.common.collect.ImmutableList;
+import com.google.common.collect.Iterables;
+import com.google.common.collect.Lists;
+import com.google.common.collect.Maps;
+import com.google.inject.Inject;
+import org.apache.hadoop.mapreduce.TaskCounter;
+
+import java.util.Collection;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+/**
+ * Retrieves information/reports for a MapReduce run via the Metrics system.
+ */
+public class MapReduceMetricsInfo {
+
+  private final MetricStore metricStore;
+
+  @Inject
+  public MapReduceMetricsInfo(MetricStore metricStore) {
+    this.metricStore = metricStore;
+  }
+
+  /**
+   * @param runId for which information will be returned.
+   * @return a {@link MRJobInfo} containing information about a particular MapReduce program run.
+   */
+  public MRJobInfo getMRJobInfo(Id.Run runId) throws Exception {
+    Preconditions.checkArgument(ProgramType.MAPREDUCE.equals(runId.getProgram().getType()));
+
+    // baseTagValues has tag keys: ns.app.mr.runid
+    List<TagValue> baseTagValues = Lists.newArrayList();
+    baseTagValues.add(new TagValue(Constants.Metrics.Tag.NAMESPACE, runId.getNamespace().getId()));
+    baseTagValues.add(new TagValue(Constants.Metrics.Tag.APP, runId.getProgram().getApplicationId()));
+    baseTagValues.add(new TagValue(Constants.Metrics.Tag.MAPREDUCE, runId.getProgram().getId()));
+    baseTagValues.add(new TagValue(Constants.Metrics.Tag.RUN_ID, runId.getId()));
+
+    List<TagValue> mapTagValues = Lists.newArrayList(baseTagValues);
+    mapTagValues.add(new TagValue(Constants.Metrics.Tag.MR_TASK_TYPE, MapReduceMetrics.TaskType.Mapper.getId()));
+
+    List<TagValue> reduceTagValues = Lists.newArrayList(baseTagValues);
+    reduceTagValues.add(new TagValue(Constants.Metrics.Tag.MR_TASK_TYPE, MapReduceMetrics.TaskType.Reducer.getId()));
+
+    // map from RunId -> (CounterName -> CounterValue)
+    Map<String, Map<String, Long>> mapTaskMetrics = Maps.newHashMap();
```

Comment:
I don't think Table is needed. This isn't too complex, (just a map from taskId to its metrics).
And Table might bring more complexity.